### PR TITLE
New package: retdec-3.2

### PR DESCRIPTION
**Pull request by marmeladema**

### CODE REVIEW

File: srcpkgs/retdec/template [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Template file for 'retdec'
+pkgname=retdec
+version=3.2
+revision=1
+patch_args="-Np1"
+build_style=cmake
+#hostmakedepends="pkg-config"
```
Comment (maxice8) [inline srcpkgs/retdec/template]: remove commented variable

File: srcpkgs/retdec/template [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+makedepends="python3 zlib-devel perl libressl-devel flex autoconf automake libtool m4 ncurses-devel ncurses-libs"
+depends="python3 upx graphviz"
+short_desc="RetDec is an open-source machine-code decompiler based on LLVM"
+maintainer="marmeladema <xademax@gmail.com>"
+license="MIT"
+homepage="https://retdec.com/"
+changelog="https://github.com/avast-tl/${pkgname}/blob/df784107fa7f062ca307b7896b80d35c1ddbe882/CHANGELOG.md"
```
[the remark this answers is not in the document]
Comment (marmeladema) [reply srcpkgs/retdec/template]: I used the changelog as documented upstream. I will remove it then.

File: srcpkgs/retdec/template [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# Template file for 'retdec'
+pkgname=retdec
+version=3.2
+revision=1
+patch_args="-Np1"
+build_style=cmake
+#hostmakedepends="pkg-config"
+makedepends="python3 zlib-devel perl libressl-devel flex autoconf automake libtool m4 ncurses-devel ncurses-libs"
```
[the remark this answers is not in the document]
Comment (abenson) [reply srcpkgs/retdec/template]: `hostmakedepends=` are things that need to be run on the host. `makedepends=` will be of the target architecture and are only linked to. `depends=` will be of the target architecture and are only installed when the package is installed. `xbps-src` will detect linked packages and automatically include them as depends. See [Manual.md](https://github.com/void-linux/void-packages/blob/master/Manual.md) for more information.

**Comment by abenson** [comment]
I [attempted to package retdec](https://github.com/abenson/void-packages-old/commit/1bce7b77c276332f4e6a421a977af6deff063c22) earlier this year. I can't remember why I didn't ultimately PR it. The resulting package was something like 4GB, and I think that was part of the reason.

File: srcpkgs/retdec/template [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# Template file for 'retdec'
+pkgname=retdec
+version=3.2
+revision=1
+patch_args="-Np1"
+build_style=cmake
+hostmakedepends="python3 perl pkg-config flex autoconf automake libtool m4"
+makedepends="zlib-devel libressl-devel ncurses-devel ncurses-libs"
```
Comment (maxice8) [inline srcpkgs/retdec/template]: ncurses-libs is not necessary

File: srcpkgs/retdec/template [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+depends="python3 upx graphviz"
+short_desc="RetDec is an open-source machine-code decompiler based on LLVM"
+maintainer="marmeladema <xademax@gmail.com>"
+license="MIT"
+homepage="https://retdec.com/"
+changelog="https://github.com/avast-tl/${pkgname}/blob/master/CHANGELOG.md"
+distfiles="https://github.com/avast-tl/${pkgname}/archive/v${version}.tar.gz"
+checksum=d1ec301a1024887431abb0fbac9478fb2cd66cbf48706174fc6a423aab4c3d60
+only_for_archs="i686 x86_64"
```
Comment (maxice8) [inline srcpkgs/retdec/template]: replace it with archs="i686 x86_64" and put directly above build_style=

File: common/shlibs [hunk 6]
```diff
@@ -3400,3 +3400,5 @@ libsnoresettings-qt5.so.0.7 snorenotify-0.7.0_1
 libTECkit.so.0 libteckit-2.5.8_1
 libTECkit_Compiler.so.0 libteckit-2.5.8_1
 libwf-config.so wf-config-0.1_1
+libretdec-libelf.so retdec-3.2_1
```
Comment (maxice8) [inline common/shlibs]: please don't add these, they seem to be internal libraries

Comment (marmeladema) [reply common/shlibs]: What do you mean by 'internal libraries'?
`retdec-bin2llvmir`, `retdec-fileinfo` and `retdec-unpacker` link against `libretdec-libdwarf.so` and `libretdec-libelf.so`.

Comment (Johnnynator) [reply common/shlibs]: They are in the same pacakge, so they don't need to be referenced in common/shlibs. A

**Comment by Johnnynator** [comment]
Can you try to get every file that get fetched into distfiles and cmake downloading files disabled?

File: common/shlibs (hunk 6 above)
Comment (marmeladema) [reply common/shlibs]: According the comment at the beginning of the shlibs file:
> Every shared library installed by a package must be listed here and mapped to a binary package.

**Comment by marmeladema** [comment]
> Can you try to get every file that get fetched into distfiles and cmake downloading files disabled?

Hum, it don't know how complicated its going to be but i will look into it.

File: common/shlibs (hunk 6 above)
Comment (marmeladema) [reply common/shlibs]: Also xbps-src complains with a warning when building the package if the libraries are not specified in shlibs. What are the rules then? It does not seem very consistent.

Comment (Johnnynator) [reply common/shlibs]: The rules are to only add an entry to common/shlibs when it is actually needed.

Comment (marmeladema) [reply common/shlibs]: Do I need to understand the rule as "if there are not a lot of chances an external binary will link against those libraries then they should _not_ be listed in shlibs" ? If yes, then shlibs comment should definitely be fixed.

**Comment by maxice8** [comment]
ping

**Pull request closed without merging**